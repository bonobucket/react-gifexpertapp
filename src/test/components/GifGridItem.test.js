import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en gif grid item', () => {

    const title = "test prueba gif grid item";
    const url = "https://localchost:4200";
    const wrapper = shallow(<GifGridItem {...{ title: title, url: url }} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img= wrapper.find('img');
       // console.log(img.props())
       expect(img.prop('src')).toBe(url)
       expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener la clase animate__fadeIn', () => {
        const cssClass= 'animate__fadeIn'
        const div= wrapper.find('div');
        const className= div.prop('className');
        expect(className.includes(cssClass)).toBe(true)
       // expect(div.hasClass('animate__fadeIn')).toBe(cssClass)
    })
    
    

})
