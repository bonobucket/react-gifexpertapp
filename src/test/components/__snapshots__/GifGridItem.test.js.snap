// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gif grid item Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test prueba gif grid item"
    src="https://localchost:4200"
  />
  <p>
    test prueba gif grid item
  </p>
</div>
`;
